apps/gallery/js/pick.js

```diff
@@ -238,9 +240,14 @@ var Pick = (function() {
       pickType = null;   // Return unchanged or convert to JPEG
     }
 
-    if (pickType && pickType !== 'image/jpeg' && pickType !== 'image/png') {
+    if (pickType && pickType !== 'image/jpeg' &&
+        pickType !== 'image/jpeg+exif' && pickType !== 'image/png') {
       pickType = null;   // Return unchanged or convert to JPEG
     }
+    else if (!pickType && pickedFileInfo.type == 'image/jpeg') {
```

Comment:
I think maybe you shouldn't have the else on this line because it won't run if you set pickType to null in the if clause above.  But note that it does run in the two other places where pickType is set to null.